test-utils/src/query/mod.rs

```diff
@@ -218,8 +218,8 @@ pub fn compare_documents(expected: &Document, actual: &Document, type_compare: b
     if expected.len() != actual.len() {
         return false;
     }
-    expected.iter().all(|(ek, ev)| {
-        actual.iter().any(|(ak, av)| {
+    expected.iter().all(|(ek, ev)| match actual.get(ek) {
```

Comment:
I think we could sort this up since each `if` clause returns the result of its evaluation when it matches. Not a required change.

```suggestion
    expected
        .iter()
        .all(|(ek, expected_value)| match actual.get(ek) {
            Some(actual_value) => match (expected_value, actual_value) {
                (Bson::Document(expected_document), Bson::Document(actual_document)) => {
                    compare_documents(expected_document, actual_document, type_compare)
                }
                (Bson::Array(expected_array), Bson::Array(actual_array)) => {
                    compare_arrays(expected_array, actual_array, type_compare)
                }
                _ if type_compare => expected_value.element_type() == actual_value.element_type(),
                _ if is_numeric(expected_value) && is_numeric(actual_value) => {
                    compare_doubles_for_test(
                        numeric_to_double(expected_value),
                        numeric_to_double(actual_value),
                    )
                }
                _ => expected_value == actual_value,
            },
            None => false,
        })
}
```

Reply:
This is a fun use of `_ <condition>` patterns

Reply:
Cool, I added this implementation.